File: Mikroservis.Clanarine/models/MemberSubscription.go
package models

import (
	"time"
)

type MemberSubscription struct {
	Id        string    `bson:"_id,omitempty"` // MongoDB koristi string kao ObjectId
	MemberId  string    `bson:"member_id"`
	Type      string    `bson:"type"` // Npr. "Monthly", "Yearly"
	Price     float64   `bson:"price"`
	StartDate time.Time `bson:"start_date"`
	EndDate   time.Time `bson:"end_date"`
	IsActive  bool      `bson:"is_active"`
}

File: Api.Gateway/Api.Gateway/Models/MemberSubscription.cs
namespace Api.Gateway.Models
{
    public class MemberSubscription
    {
        public string Id { get; set; } // Identifikator pretplate
        public string MemberId { get; set; } // ID člana na kojeg se odnosi pretplata
        public string Type { get; set; } // Tip pretplate, npr. "Monthly", "Yearly"
        public decimal Price { get; set; } // Cena pretplate
        public DateTime StartDate { get; set; } // Datum početka pretplate
        public DateTime EndDate { get; set; } // Datum završetka pretplate
        public bool IsActive { get; set; } // Status pretplate
    }
}
